From b833ecb647e21b614335eae826587f19ac99e01a Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Tue, 26 Dec 2017 14:48:50 +0800
Subject: [PATCH] UPSTREAM: media: cec: cec_transmit_attempt_done: ignore 
 CEC_TX_STATUS_MAX_RETRIES

The switch in cec_transmit_attempt_done() should ignore the
CEC_TX_STATUS_MAX_RETRIES status bit.

Calling this function with e.g. CEC_TX_STATUS_NACK | CEC_TX_STATUS_MAX_RETRIES
is perfectly legal and should not trigger the WARN(1).

Signed-off-by: Hans Verkuil <hans.verkuil@cisco.com>
Signed-off-by: Mauro Carvalho Chehab <mchehab@s-opensource.com>
(cherry picked from commit bd34ca8758d0ee9b13c0c067a956b0e9eb247d18)

Change-Id: I0db57343d0a922b2154e8b10979b226f116b750c
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/media/cec/cec-adap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/cec/cec-adap.c b/drivers/media/cec/cec-adap.c
index 1a021828c8d4..8ac39ddf892c 100644
--- a/drivers/media/cec/cec-adap.c
+++ b/drivers/media/cec/cec-adap.c
@@ -609,7 +609,7 @@ EXPORT_SYMBOL_GPL(cec_transmit_done_ts);
 void cec_transmit_attempt_done_ts(struct cec_adapter *adap,
 				  u8 status, ktime_t ts)
 {
-	switch (status) {
+	switch (status & ~CEC_TX_STATUS_MAX_RETRIES) {
 	case CEC_TX_STATUS_OK:
 		cec_transmit_done_ts(adap, status, 0, 0, 0, 0, ts);
 		return;
-- 
2.35.3

